indico/htdocs/js/indico/modules/contributions/common.js

```diff
@@ -28,11 +28,80 @@
         });
     }
 
+    function formatState(visible, total) {
+        return $T.gettext('{0} / {1}').format('<strong>{0}</strong>'.format(visible.length),
+                                              total.length);
+    }
+
+    function setState(visible, total) {
+        var $state = $('#filtering-state'),
+            title = format('{0} out of {1} contributions displayed', [visible.length, total.length]);
+        $state.html(formatState(visible, total));
+
+        // oldtitle needs to be updated too, because of qTip
+        $state.attr({
+            oldtitle: title,
+            title: title
+        });
+    }
+
+    function applyFilters() {
+        var contributions = $('#contribution-list tbody tr'),
+            term = $('#search-input').val().trim(),
+            visibleEntries, m,
+            $state = $('#filtering-state'),
+            $filterPlaceholder = $('#filter-placeholder');
+
+        $filterPlaceholder.hide();
+        $state.removeClass('active');
+
+        if (term == '') {
+            contributions.show();
+            setState(contributions, contributions);
+            return;
+        }
+
+        // quick search of contribution by ID
+        if (m = term.match(/^#\d+$/)) {
+            visibleEntries = contributions.find('.id-column').filter(function() {
+                return $(this).text().toLowerCase() == m[0];
+            }).closest('tr');
+        } else {
+            visibleEntries = contributions.find('td[data-searchable*=' + term + ']').closest('tr');
```

Comment:
the term should be quoted and escaped; otherwise it'll break whne using e.g. `]` in it (or `"` when it's quoted but not escaped)